CRUD2-Hibernate/src/main/java/Servlets/EditServlet.java

```
@@ -0,0 +1,52 @@
+package Servlets;
+
+import models.Employee;
+import services.CRUDService;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/edit")
+public class EditServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        CRUDService service = new CRUDService();
+
+        try {
+            int id = Integer.parseInt(req.getParameter("id"));
+            Employee employee = service.findEmployeeById(id);
+            if (employee != null) {
+                req.setAttribute("employee", employee);
+                getServletContext().getRequestDispatcher("/edit.jsp").forward(req, resp);
+            } else {
+                getServletContext().getRequestDispatcher("/notFound.jsp").forward(req, resp);
```

DRY. Код else идентичен коду в catch, т.е. ты можешь здесь сделать throw и получить тот же результат. Чтобы ещё подсократить метод и следовать fail fast идеологии, можно заменить проверку на наличие проверкой на отсутствии ( к тому же, равенство всегда проще читается, чем неравенство), так:

```
 if (employee == null) {

    throw new Exception("your exception string")

}
req.setAttribute ....
```

---

Исправлено